app/controllers/friendings_controller.rb

```diff
@@ -0,0 +1,25 @@
+class FriendingsController < ApplicationController
+
+  def create
+    new_friend = User.find(params[:format])
+    if current_user.friendeds << new_friend
+      flash.notice = "Successfully friended #{new_friend.profile.name}"
+      redirect_back(fallback_location: current_user)
+    else
+      flash.notice = "Failed to friend!  Sad :("
+      redirect_back(fallback_location: current_user)
+    end
+  end
+
+  def destroy
```

Comment:
You could probably add a conditional here in case you're unable to "unfriend" for some reason.